recipes/sail/all/conanfile.py

```diff
@@ -60,73 +80,127 @@ def configure(self):
         if self.options.shared:
             del self.options.fPIC
 
+    def _expand_high_priority(self):
+        self.options.with_bmp = True
+        self.options.with_gif = True
+        # Don't overwrite the value set by the user
+        if not self.options.with_jpeg:
+            self.options.with_jpeg = "libjpeg-turbo"
+        self.options.with_png = True
+        self.options.with_tiff = True
+
+    def _expand_medium_priority(self):
+        self.options.with_avif = True
+        self.options.with_jpeg2000 = True
+        self.options.with_webp = True
+
+    def _expand_low_priority(self):
+        self.options.with_ico = True
+        self.options.with_pcx = True
+        self.options.with_qoi = True
+        self.options.with_tga = True
+
+    def _expand_lowest_priority(self):
+        self.options.with_wal = True
+        self.options.with_xbm = True
+
     def requirements(self):
+        if self.options.with_all:
+            self._expand_high_priority()
+            self._expand_medium_priority()
+            self._expand_low_priority()
+            self._expand_lowest_priority()
+        else:
+            if self.options.with_high_priority:
+                self._expand_high_priority()
+            if self.options.with_medium_priority:
+                self._expand_medium_priority()
+            if self.options.with_low_priority:
+                self._expand_low_priority()
+            if self.options.with_lowest_priority:
+                self._expand_lowest_priority()
+
```

Comment:
this logic must not live in requirements(), and I don't think it's a good recipe design anyway.

Reply:
I think I'll remove for now